src/NetAnalyzers/Core/Microsoft.NetCore.Analyzers/Performance/UseStringMethodCharOverloadWithSingleCharacters.cs

```diff
@@ -0,0 +1,239 @@
+
+            if (!stringTypeHasCharOverload || stringComparisonType == null || cultureInfoType == null)
+                return;
+
+            var stringComparisonOrdinalSymbol = stringComparisonType.GetMembers(nameof(StringComparison.Ordinal)).First();
+            var stringComparisonInvariantCultureSymbol = stringComparisonType.GetMembers(nameof(StringComparison.InvariantCulture)).First();
+            var cultureInfoInvariantCultureSymbol = cultureInfoType.GetMembers(nameof(CultureInfo.InvariantCulture)).First();
+
+            context.RegisterOperationAction(
+                context => AnalyzeOperation(
+                    context,
+                    stringComparisonType,
+                    cultureInfoType,
+                    stringComparisonOrdinalSymbol,
+                    stringComparisonInvariantCultureSymbol,
+                    cultureInfoInvariantCultureSymbol),
+                OperationKind.Invocation);
+        }
+
+        private void AnalyzeOperation(
+            OperationAnalysisContext context,
+            INamedTypeSymbol stringComparisonType,
+            INamedTypeSymbol cultureInfoType,
+            ISymbol stringComparisonOrdinalSymbol,
+            ISymbol stringComparisonInvariantCultureSymbol,
+            ISymbol cultureInfoInvariantCultureSymbol)
+        {
+            var invocationOperation = (IInvocationOperation)context.Operation;
+            if (TryMatchTargetMethod(invocationOperation, out var method, out var comparison) &&
+                TryGetChar(invocationOperation, out var stringArgument, out var c))
+            {
+                DiagnosticDescriptor? rule;
+
+                // CA1865: Method(string, StringComparison.Ordinal)
+                //         Method(printable ascii string, StringComparison.InvariantCulture)
+                //         Method(printable ascii string, false/true, CultureInfo.InvariantCulture)
+                if (comparison == ComparisonUsed.Ordinal ||
+                    (comparison == ComparisonUsed.InvariantCulture && c.IsPrintableAscii()))
+                {
+                    rule = SafeTransformationRule;
+                }
+                // CA1866: Method(string)
+                else if (comparison == null)
+                {
+                    rule = NoSpecifiedComparisonRule;
+                }
+                // CA1867: Method(string, StringComparison.AnythingElse)
+                //         Method(string, false/true, CultureInfo.AnythingElse)
+                else
+                {
+                    rule = AnyOtherSpecifiedComparisonRule;
+                }
+
+                var argumentList = GetArgumentList(stringArgument.Syntax);
+                if (argumentList != null)
+                {
+                    var arg0 = $"string.{method}(char)";
+                    var arg1 = $"string.{method}(string)";
+                    context.ReportDiagnostic(argumentList.CreateDiagnostic(rule, new[] { arg0, arg1 }));
```

Comment:
Should the message be updated to `Use 'string.{0}(char)' instead of 'string.{0}(string)'` then we just pass "method" as the single argument?

Reply:
Sure. I kept them separate in case these args need to change based on the selected method overload (for example if it's `string.{method}(string)` vs `string.{method}(string, StringComparison)`), but I guess we don't need to do that.